jwplatform/client.py

```diff
@@ -219,28 +240,186 @@ def clip(self, site_id, channel_id, event_id, body=None, query_params=None):
         )
 
 
-class _MediaClient(_SiteResourceClient):
+CREATE_MEDIA_PAYLOAD = {
+    "upload": {
+    },
+    "metadata": {
 
+    }
+}
+
+
+class _MediaClient(_SiteResourceClient):
     _resource_name = "media"
     _id_name = "media_id"
+    _logger = logging.getLogger(__name__)
 
     def reupload(self, site_id, body, query_params=None, **kwargs):
         resource_id = kwargs[self._id_name]
         return self._client.request(
             method="PUT",
-            path=self._singular_path.format(site_id=site_id, resource_name=self._resource_name, resource_id=resource_id) + "reupload/",
+            path=self._singular_path.format(site_id=site_id, resource_name=self._resource_name,
+                                            resource_id=resource_id) + "reupload/",
             body=body,
             query_params=query_params
         )
 
-class _WebhookClient(_ResourceClient):
+    def _determine_upload_method(self, file, target_part_size) -> str:
+        file_size = os.stat(file.name).st_size
+        if file_size > MAX_FILE_SIZE:
+            raise NotImplementedError('File size greater than 25 GB is not supported.')
+        if file_size <= target_part_size:
+            return UploadType.direct.value
+        return UploadType.multipart.value
+
+    def create_media_and_get_upload_context(self, file, body=None, query_params=None, **kwargs) -> UploadContext:
+        """
+        Creates the media and retrieve the upload context
+        Args:
+            file: The file-like object to the actual media file to be uploaded
+            body: The body of the payload.
+            query_params: The query parameters.
+            **kwargs: The upload arguments.
+
+        Returns: The UploadContext that can be reused to resuming an upload.
+
+        """
+        if not kwargs:
+            kwargs = {}
+        site_id = kwargs['site_id']
+        # Determine the upload type - Single or multi-part
+        target_part_size = int(kwargs['target_part_size']) if 'target_part_size' in kwargs else MIN_PART_SIZE
```

Comment:
```suggestion
        target_part_size = int(kwargs.get('target_part_size', MIN_PART_SIZE))
```
Similar to other comments, I think this is more idiomatic to python.

Reply:
done.